### PR TITLE
Low hanging fruit performance improvements from profiling session

### PR DESCRIPTION
**Pull request by Praburaj**
This PR addresses the below bullet from this bug: https://github.com/aspnet/XRE/issues/1147

Don't use Tracing system for logging, use Console.WriteLine for now. Trace loads the entire configuration stack.

@davidfowl

### CODE REVIEW

File: src/Microsoft.Framework.Runtime/Project.cs [hunk 1]
```diff
@@ -443,7 +443,7 @@ private static IEnumerable<string> GetSourcePatternCore(JObject rawProject, stri
             }
 
             // Assume it's an array (it should explode if it's not)
-            return token.ToObject<string[]>().SelectMany(GetSourcesSplit);
+            return token.Select(t => t.Value<string>()).SelectMany(GetSourcesSplit);
```
Comment (davidfowl) [inline src/Microsoft.Framework.Runtime/Project.cs]: Lets add an extension method, ValueAsArray<T>:

```C#
script.Value.Select(s => s.Value<string>()).ToArray()
```
```C#
script.ValueAsArray<string>()
```

Comment (davidfowl) [inline src/Microsoft.Framework.Runtime/Project.cs]: And use it everywhere there's a ```ToObject<string[]>```

File: src/Microsoft.Framework.DesignTimeHost/ApplicationContext.cs [hunk 2]
```diff
@@ -235,7 +242,14 @@ private bool ProcessMessage()
                     break;
                 case "GetCompiledAssembly":
                     {
-                        var libraryKey = message.Payload.ToObject<RemoteLibraryKey>();
```
Comment (davidfowl) [inline src/Microsoft.Framework.DesignTimeHost/ApplicationContext.cs]: Delete the ```RemoteLibraryKey``` class

File: src/Microsoft.Framework.DesignTimeHost/ApplicationContext.cs [hunk 3]
```diff
@@ -989,6 +1003,23 @@ private static DependencyDescription CreateDependencyDescription(LibraryDescript
             };
         }
 
+        private static T GetValue<T>(JToken token, string name)
+        {
+            if (token == null)
```
Comment (davidfowl) [inline src/Microsoft.Framework.DesignTimeHost/ApplicationContext.cs]: This can be C# 6 ized:

```C#
token?[name]?.Value<T>()
```
Ninjaaaaa

Comment (Praburaj) [inline src/Microsoft.Framework.DesignTimeHost/ApplicationContext.cs]: yeah. I'll fix that.

File: src/Microsoft.Framework.Runtime/Project.cs (hunk 1 above)
Comment (Praburaj) [inline src/Microsoft.Framework.Runtime/Project.cs]: Sure. I'll update the PR with this feedback.

**Comment by Praburaj** [comment]
@davidfowl Updated PR.

File: src/Microsoft.Framework.Runtime/GlobalSettings.cs [hunk 4]
```diff
@@ -57,7 +57,9 @@ public static bool TryGetGlobalSettings(string path, out GlobalSettings globalSe
             var projectSearchPaths = settings["projects"] ?? settings["sources"];
             var dependencies = settings["dependencies"] as JObject;
 
-            globalSettings.ProjectSearchPaths = projectSearchPaths == null ? new string[] { } : projectSearchPaths.ToObject<string[]>();
+            globalSettings.ProjectSearchPaths = projectSearchPaths == null ?
+                new string[] { } :
+                projectSearchPaths.Select(p => p.Value<string>()).ToArray();
```
Comment (davidfowl) [inline src/Microsoft.Framework.Runtime/GlobalSettings.cs]: ValueAsArray()

Comment (Praburaj) [inline src/Microsoft.Framework.Runtime/GlobalSettings.cs]: Ah.. fixed.

File: src/Microsoft.Framework.Runtime/JTokenExtensions.cs [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+using System.Linq;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.Framework.Runtime
+{
+    public static class JTokenExtensions
+    {
+        public static T[] ValueAsArray<T>(this JToken jToken)
+        {
+            return jToken.Select(a => a.Value<T>()).ToArray();
```
Comment (muratg) [inline src/Microsoft.Framework.Runtime/JTokenExtensions.cs]: need a null check here?

**Comment by muratg** [comment]
:shipit:

**Comment by davidfowl** [comment]
:shipit:

File: src/Microsoft.Framework.Runtime/JTokenExtensions.cs (hunk 5 above)
Comment (Praburaj) [inline src/Microsoft.Framework.Runtime/JTokenExtensions.cs]: All callers handle the null here. But it wouldn't hurt adding a null check.

**Comment by Praburaj** [comment]
merged : https://github.com/aspnet/XRE/commit/f748be51c055b1b136888bb32a595b29f47fac7f

**Pull request closed without merging**